code/modules/mob/living/simple_animal/revenant/revenant.dm

```diff
@@ -370,36 +378,41 @@
 
 /obj/item/weapon/ectoplasm/revenant/examine(mob/user)
 	..()
-	if(reforming)
-		user << "<span class='warning'>It is shifting and distorted. It would be wise to destroy this.</span>"
-	else if(!reforming)
-		user << "<span class='notice'>It seems inert.</span>"
+	if(inert)
+		user << "<span class='revennotice'>It seems inert.</span>"
+	else if(reforming)
+		user << "<span class='revenwarning'>It is shifting and distorted. It would be wise to destroy this.</span>"
 
 /obj/item/weapon/ectoplasm/revenant/proc/reform()
-	if(!reforming || !src)
+	if(inert || !src)
 		return
 	var/key_of_revenant
-	message_admins("Revenant ectoplasm was left undestroyed for 1 minute and has reformed into a new revenant.")
+	message_admins("Revenant ectoplasm was left undestroyed for 1 minute and is reforming into a new revenant.")
 	loc = get_turf(src) //In case it's in a backpack or someone's hand
-	visible_message("<span class='boldannounce'>[src] suddenly rises into the air before fading away.</span>")
 	var/mob/living/simple_animal/revenant/R = new(get_turf(src))
 	if(client_to_revive)
 		for(var/mob/M in mob_list)
-			if(M.client == client_to_revive && M.stat == DEAD) //Only recreates the mob if the mob the client is in is dead
-				message_admins("[R.client] was a revenant and died. Re-making them into the new revenant formed by ectoplasm.")
+			if(M.stat == DEAD && M.client == client_to_revive) //Only recreates the mob if the mob the client is in is dead
+				message_admins("[M.client] was a revenant and died. Re-making them into the new revenant formed by ectoplasm.")
 				R.client = client_to_revive
 				key_of_revenant = client_to_revive.key
 				break
```

Comment:
This chunk, 394 to 399, doesn't fucking work right please inform me how to fix this and or where Xhuis is so I can punch his face.
Specifically, even if the client is in a non-dead mob, it'll still show the admin message and fail to try and get other candidates.